src/HaskellCI.hs

```diff
@@ -157,13 +159,17 @@ travisFromConfigFile args opts path = do
         | "cabal.project" `isPrefixOf` takeFileName path = Just path
         | otherwise = Nothing
 
-    getCabalFiles :: m (Project FilePath)
+    getCabalFiles :: m (Project Void FilePath)
     getCabalFiles
         | isNothing isCabalProject = return $ emptyProject & #prjPackages .~ [path]
         | otherwise = do
             contents <- liftIO $ BS.readFile path
-            pkgs <- either putStrLnErr return $ parseProjectFile path contents
-            over #prjPackages concat `liftM` T.mapM findProjectPackage pkgs
+            prj  <- either putStrLnErr return $ parseProjectFile path contents
+            prj' <- bitraverse findOptProjectPackage findProjectPackage prj
+            return prj'
+                { prjPackages    = concat $ prjPackages prj' ++ prjOptPackages prj'
```

Comment:
It feels a bit weird to take both the `prjPackages` and `prjOptPackages` from one `Project` and combine them into a _single_ `prjPackages` list in a new `Project`, leaving the new `prjOptPackages` field empty. Is there a reason things are done this way?

Reply:
yes. The only difference between`optional-packages` and `packages` is that former "globs" can match zero packages, but when the `.travis.yml` is generated we know for sure which packages should be there.

If someone removes or ads an optional package, they have to run the generation script ianyway, as it does resolving (because we want to `sdist` first).

https://cabal.readthedocs.io/en/latest/nix-local-build.html#cfg-field-optional-packages